### PR TITLE
Reduce inheritance, fix #91

### LINKED ISSUE #91: Q: Why `Helper::Row::SelfResultSet` disables `InflateColumn::DateTime` if used first?
When I load components at this order:

	Helper::Row::SelfResultSet
	InflateColumn::DateTime

The DateTime fields are not deflated:

At point: https://metacpan.org/source/RIBASUSHI/DBIx-Class-0.082841/lib/DBIx/Class/Row.pm#L407

the $self object is:
```
DBG>$self
HyperMouse::Schema::Result::Contractor {
  _column_data => {
    contractor_type_id => 1,
    known_from => DateTime 2018-07-13T13:18:24,
    known_till => DateTime 9999-12-31T23:59:59,
    name => Contractor,
    provider => 1,
    valid_from => DateTime 2018-07-13T13:18:24,
    valid_till => DateTime 9999-12-31T23:59:59,
  },
  _in_storage => 0,
  _result_source => IGNORED,
}
```

In compare when `Helper::Row::SelfResultSet` is last loaded component:

```
HyperMouse::Schema::Result::Contractor {
  _column_data => {
    contractor_type_id => 1,
    name => Contractor,
    provider => 1,
  },
  _in_storage => 0,
  _inflated_column => {
    known_from => DateTime 2018-07-13T13:19:47,
    known_till => DateTime 9999-12-31T23:59:59,
    valid_from => DateTime 2018-07-13T13:19:47,
    valid_till => DateTime 9999-12-31T23:59:59,
  },
  _rel_in_storage => {},
  _result_source => IGNORED,
}
```

### PR DESCRIPTION
**Pull request by dakkar**
I have removed most `use parent`, and replaced the necessary ones with the smallest needed class.

I've also added a test showing the problem reported in issue #91. I think the root of the problem is that derived classes that don't call `load_components` don't get the C3 mro.

All tests still pass on my machine.

### CODE REVIEW

**Comment by dakkar** [comment]
Yep, confirmed, adding `use mro 'c3'` to the derived classes in `t/component-order.t` makes it pass without my patches. I still think that it's a good idea to reduce the inheritance tree, though.

File: lib/DBIx/Class/Helper/ResultSet/IgnoreWantarray.pm [hunk 1]
```diff
@@ -5,9 +5,7 @@ package DBIx::Class::Helper::ResultSet::IgnoreWantarray;
 use strict;
 use warnings;
 
-use parent 'DBIx::Class::ResultSet';
-
-sub search :DBIC_method_is_indirect_sugar{
```
Comment (frioux) [inline lib/DBIx/Class/Helper/ResultSet/IgnoreWantarray.pm]: Why did you remove this attribute?  Is this no longer needed?

**Comment by frioux** [comment]
Unless you can show a concrete improvement from this patchset I'm inclined to close it.  This is what DBIx::Class::Helper::Schema::Verifier::C3 was built to automate: people *must* use c3, one way or another, with DBIC.

File: lib/DBIx/Class/Helper/ResultSet/IgnoreWantarray.pm (hunk 1 above)
Comment (dakkar) [reply lib/DBIx/Class/Helper/ResultSet/IgnoreWantarray.pm]: I grepped for it in the source code of the latest released DBIC, and couldn't find it, so my tests failed. I don't know how they pass for you and CI, I suspect I'm missing something crucial.

**Comment by dakkar** [comment]
> people must use c3, one way or another, with DBIC

Very fair! Maybe add a line of documentation in https://github.com/frioux/DBIx-Class-Helpers/blob/master/lib/DBIx/Class/Helper/ResultSet.pm and to a Result helper or doc-only package, explaining that?

One of the problems is that Schema::Loader puts a `use base` in the generated Result classes, and no `use mro 'c3'`, even whed a custom base class is specified (that's how I encountered this issue). Should I ask the Schema::Loader maintaner to add that?

**Comment by frioux** [comment]
yeah they either need to use mro 'c3' or *I think* calling load_components does that for you as well.

**Comment by dakkar** [comment]
I reported the suggestion to Schema::Loader https://rt.cpan.org/Ticket/Display.html?id=132035

**Comment by frioux** [comment]
Cool.  Gonna close this one now.  Thanks!
